examples/Nat/IProtocolHelper.cs

```diff
@@ -0,0 +1,40 @@
+using PacketDotNet;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Pax.Examples.Nat
+{
+  internal interface IProtocolHelper<T> where T : Packet
+  {
+    /// <summary>
+    /// Gets an initial state for this Transport-layer protocol.
+    /// </summary>
+    /// <returns>An initial state.</returns>
+    ITransportState<T> InitialState();
+  }
+
+  internal sealed class GenericTransportHelper<T> : IProtocolHelper<T> where T : Packet
+  {
+    public ITransportState<T> InitialState()
+    {
+      return NoTransportState<T>.Instance;
+    }
+  }
+
+  internal sealed class TcpHelper : IProtocolHelper<TcpPacket>
```

Comment:
The number of classes is proliferating :) Could you add comments that describe the role of each class please?